composeApp/src/commonMain/kotlin/com/github/csc3380fall2024/team16/ui/routes/root/authenticated/social/SocialScreen.kt

```diff
@@ -10,37 +11,58 @@ import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccountCircle
+import androidx.compose.material.icons.filled.Add
 import androidx.compose.material3.Button
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
 import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextField
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.graphics.painter.Painter
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 
 @Composable
-fun SocialScreen() {
-    var showProfileCreation by remember { mutableStateOf(true) }
-    var name by remember { mutableStateOf("") }
-    var bio by remember { mutableStateOf("") }
-    var profilePicture: Painter? by remember { mutableStateOf(null) } // Placeholder for image handling
+fun SocialScreen(viewModel: SocialViewModel) {
+    var showProfileCreation by rememberSaveable { mutableStateOf(viewModel.name.isNullOrEmpty() && viewModel.bio.isNullOrEmpty()) }
+    var name by rememberSaveable { mutableStateOf(viewModel.name ?: "") }
+    var bio by rememberSaveable { mutableStateOf(viewModel.bio ?: "") }
+    var profilePicture by rememberSaveable { mutableStateOf(viewModel.profilePicture) }
```

Comment:
Why are you using remember and creating mutable states for viewmodel fields? The fields themselves are already mutable states